### PR TITLE
delete listeners and view reference in region app when destroyed

### PR DESCRIPTION
**Pull request by wesmangum**
Why: Destroying a region's view is not correctly deleting any references or listeners to the view.

What: Each instance of a view being destroyed needs to ensure that listeners and references are removed as well.

How: In `setView`, `_regionEventMonitor`,  and `destroy`, there is now a check to ensure `stopListening` is called on the view and that it's deleted.

Coverage Report: 
```
------------------------|----------|----------|----------|----------|----------------|
File                    |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
------------------------|----------|----------|----------|----------|----------------|
 src/                   |      100 |      100 |      100 |      100 |                |
  app.js                |      100 |      100 |      100 |      100 |                |
  component.js          |      100 |      100 |      100 |      100 |                |
  marionette.toolkit.js |      100 |      100 |      100 |      100 |                |
 src/mixins/            |      100 |      100 |      100 |      100 |                |
  child-apps.js         |      100 |      100 |      100 |      100 |                |
  event-listeners.js    |      100 |      100 |      100 |      100 |                |
  state.js              |      100 |      100 |      100 |      100 |                |
  view-events.js        |      100 |      100 |      100 |      100 |                |
------------------------|----------|----------|----------|----------|----------------|
All files               |      100 |      100 |      100 |      100 |                |
------------------------|----------|----------|----------|----------|----------------|


=============================== Coverage summary ===============================
Statements   : 100% ( 316/316 )
Branches     : 100% ( 132/132 )
Functions    : 100% ( 80/80 )
Lines        : 100% ( 311/311 )
================================================================================
```

I've written test cases for a view being emptied from the region and independently destroyed, but there might be edge cases I'm missing. Any feedback or test suggestions would be appreciated.

### CODE REVIEW

**Comment by wesmangum** [comment]
Closes #216

File: src/app.js [hunk 1]
```diff
@@ -342,6 +345,31 @@ const App = Marionette.Application.extend({
   },
 
   /**
+   * Region monitor handler which empties the region's view
+   *
+   * @private
+   * @method _onEmpty
+   * @memberOf App
+   */
+  _onEmpty(region, view) {
```
Comment (paulfalgout) [inline src/app.js]: I think you can remove the args here since neither are in use.
